src/Frontend/Modules/ContentBlocks/Widgets/Detail.php

```diff
@@ -23,12 +23,20 @@ public function execute()
             Locale::frontendLanguage()
         );
 
+        // Init template
+        $template = "Default.html.twig";
+
         // if the content block is not found or if it is hidden, just return an array with empty text
         // @deprecated fix this for version 5, we just shouldn't assign this instead of this hack, but we need it for BC
         if (!$contentBlock instanceof ContentBlock || $contentBlock->isHidden()) {
             $contentBlock = ['text' => ''];
+        } else {
+            $template = $contentBlock->getTemplate();
         }
 
+        // Load template
+        $this->loadTemplate('ContentBlocks/Layout/Widgets/' . $template);
```

Comment:
No, that's not true, since if you give a wrong "ID" you don't have a ContentBlock Entity.